libs/elasticsearch-core/src/test/java/org/elasticsearch/core/internal/io/IOUtilsTests.java

```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    public void testDeleteFilesIgnoringExceptionsArray() throws IOException {
+        runDeleteFilesIgnoringExceptionsTest(Function.identity(), IOUtils::deleteFilesIgnoringExceptions);
+    }
+
+    public void testDeleteFilesIgnoringExceptionsIterable() throws IOException {
+        runDeleteFilesIgnoringExceptionsTest(Arrays::asList, IOUtils::deleteFilesIgnoringExceptions);
+    }
+
+    private <T> void runDeleteFilesIgnoringExceptionsTest(
+            final Function<Path[], T> function, CheckedConsumer<T, IOException> deleteFilesIgnoringExceptions) throws IOException {
+        final int numberOfFiles = randomIntBetween(0, 7);
+        final Path[] files = new Path[numberOfFiles];
+        for (int i = 0; i < numberOfFiles; i++) {
+            if (randomBoolean()) {
+                files[i] = createTempFile();
+            } else {
+                final Path temporary = createTempFile();
+                files[i] = PathUtils.get(temporary.toString(), randomAlphaOfLength(8));
+                Files.delete(temporary);
+            }
+        }
+        deleteFilesIgnoringExceptions.accept(function.apply(files));
+        for (int i = 0; i < numberOfFiles; i++) {
+            assertFalse(files[i].toString(), Files.exists(files[i]));
+        }
+    }
+
+    public void testRm() throws IOException {
+        final int numberOfLocations = randomIntBetween(0, 7);
+        final Path[] locations = new Path[numberOfLocations];
+        for (int i = 0; i < numberOfLocations; i++) {
+            locations[i] = createTempDir();
+            Path location = locations[i];
+            while (true) {
+                location = Files.createDirectory(location.resolve(randomAlphaOfLength(8)));
+                if (rarely() == false) {
+                    Files.createTempFile(location, randomAlphaOfLength(8), null);
+                    break;
+                }
+            }
+        }
+
+        IOUtils.rm(locations);
+
+        for (int i = 0; i < numberOfLocations; i++) {
+            assertFalse(locations[i].toString(), Files.exists(locations[i]));
+        }
+    }
+
+    public void testFsyncDirectory() throws Exception {
+        final Path path = createTempDir().toRealPath();
+        final Path subPath = path.resolve(randomAlphaOfLength(8));
+        Files.createDirectories(subPath);
+        IOUtils.fsync(subPath, true);
+        // no exception
+    }
+
+    public void testFsyncFile() throws IOException {
+        final Path path = createTempDir().toRealPath();
+        final Path subPath = path.resolve(randomAlphaOfLength(8));
+        Files.createDirectories(subPath);
+        final Path file = subPath.resolve(randomAlphaOfLength(8));
+        try (OutputStream o = Files.newOutputStream(file)) {
+            o.write("0\n".getBytes(StandardCharsets.US_ASCII));
+        }
+        IOUtils.fsync(file, false);
+        // no exception
```

Comment:
This seems like as super critical thing not to have a stronger assertion about. I think it is worth brainstorming a better test for a followup.